core/common/src/test/java/alluxio/CommonTestUtils.java

```diff
@@ -54,4 +57,29 @@
     T finalObject = (T) current;
     return finalObject;
   }
+
+  /**
+   * Waits for a condition to be satisfied until a timeout occurs.
+   *
+   * @param condition the condition to wait on
+   * @param timeoutMs the number of milliseconds to wait before giving up and throwing an exception
+   */
+  public static void waitFor(Function<Void, Boolean> condition, int timeoutMs) {
+    long start = System.currentTimeMillis();
+    while (!condition.apply(null)) {
+      if (System.currentTimeMillis() - start > timeoutMs) {
+        throw new RuntimeException("Timed out waiting for condition " + condition);
+      }
+      CommonUtils.sleepMs(20);
+    }
+  }
+
+  /**
+   * Waits for 5 seconds for the condition to be satisfied.
+   *
+   * @param condition the condition to wait on
+   */
+  public static void waitFor(Function<Void, Boolean> condition) {
```

Comment:
Is this used? I think the model is to have a `waitFor(timeout)`, and a `waitFor()` that waits forever. Do we even want one that waits forever?